benchmark/torch/sac/train.py

```diff
@@ -0,0 +1,150 @@
+        if rpm.size() >= WARMUP_STEPS:
+            batch_obs, batch_action, batch_reward, batch_next_obs, batch_terminal = rpm.sample_batch(
+                BATCH_SIZE)
+            agent.learn(batch_obs, batch_action, batch_reward, batch_next_obs,
+                        batch_terminal)
+
+    return episode_reward, episode_steps
+
+
+# Runs policy for 5 episodes by default and returns average reward
+# A fixed seed is used for the eval environment
+def run_evaluate_episodes(agent, env, eval_episodes):
+    avg_reward = 0.
+    for _ in range(eval_episodes):
+        obs = env.reset()
+        done = False
+        while not done:
+            action = agent.predict(obs)
+            obs, reward, done, _ = env.step(action)
+            avg_reward += reward
+    avg_reward /= eval_episodes
+    return avg_reward
+
+
+def main():
+    logger.info("------------------- SAC ---------------------")
+    logger.info('Env: {}, Seed: {}'.format(args.env, args.seed))
+    logger.info("---------------------------------------------")
+
+    env = gym.make(args.env)
+    env.seed(args.seed)
+    env = ActionMappingWrapper(env)
+
+    obs_dim = env.observation_space.shape[0]
+    action_dim = env.action_space.shape[0]
+
+    # Initialize model, algorithm, agent, replay_memory
+    model = MujocoModel(obs_dim, action_dim)
+    algorithm = SAC(
+        model,
+        gamma=GAMMA,
+        tau=TAU,
+        alpha=ALPHA,
+        actor_lr=ACTOR_LR,
+        critic_lr=CRITIC_LR)
+    agent = MujocoAgent(algorithm)
+    rpm = ReplayMemory(
+        max_size=MEMORY_SIZE, obs_dim=obs_dim, act_dim=action_dim)
+
+    total_steps = 0
+    test_flag = 0
+    while total_steps < args.max_timesteps:
+        # Train episode
+        episode_reward, episode_steps = run_train_episode(agent, env, rpm)
+        total_steps += episode_steps
+
+        tensorboard.add_scalar('train/episode_reward', episode_reward,
+                               total_steps)
+        logger.info('Total Steps: {} Reward: {}'.format(
+            total_steps, episode_reward))
+
+        # Evaluate episode
+        if (total_steps + 1) // EVAL_EVERY_STEPS >= test_flag:
+            while (total_steps + 1) // EVAL_EVERY_STEPS >= test_flag:
+                test_flag += 1
+            avg_reward = run_evaluate_episodes(agent, env, EVAL_EPISODES)
+            tensorboard.add_scalar('eval/episode_reward', avg_reward,
+                                   total_steps)
+            logger.info('Evaluation over: {} episodes, Reward: {}'.format(
+                EVAL_EPISODES, avg_reward))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--env", default="HalfCheetah-v1", help='Mujoco gym environment name')
+    parser.add_argument(
+        "--seed",
+        default=0,
+        type=int,
+        help='Sets Gym, PyTorch and Numpy seeds')
+    parser.add_argument(
+        "--max_timesteps",
```

Comment:
please replace `max_timesteps` by `train_total_steps`, and add `test_every_steps` in add_argument, refer https://github.com/PaddlePaddle/PARL/blob/develop/examples/SAC/train.py#L139